azure/scope/managedcontrolplane_test.go

```diff
@@ -267,3 +267,135 @@ func TestManagedControlPlaneScope_AddonProfiles(t *testing.T) {
 		})
 	}
 }
+
+func TestManagedControlPlaneScope_OSType(t *testing.T) {
+	scheme := runtime.NewScheme()
+	_ = capiv1exp.AddToScheme(scheme)
+	_ = infrav1.AddToScheme(scheme)
+
+	cases := []struct {
+		Name     string
+		Input    ManagedControlPlaneScopeParams
+		Expected []azure.AgentPoolSpec
+		Err      string
+	}{
+		{
+			Name: "with Linux and Windows pools",
+			Input: ManagedControlPlaneScopeParams{
+				AzureClients: AzureClients{
+					Authorizer: autorest.NullAuthorizer{},
+				},
+				Cluster: &clusterv1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "cluster1",
+						Namespace: "default",
+					},
+				},
+				ControlPlane: &infrav1.AzureManagedControlPlane{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "cluster1",
+						Namespace: "default",
+					},
+					Spec: infrav1.AzureManagedControlPlaneSpec{
+						Version:        "v1.20.1",
+						SubscriptionID: "00000000-0000-0000-0000-000000000000",
+					},
+				},
+				ManagedMachinePools: []ManagedMachinePool{
+					{
+						MachinePool:      getMachinePool("pool0"),
+						InfraMachinePool: getAzureMachinePool("pool0", infrav1.NodePoolModeSystem),
+					},
+					{
+						MachinePool:      getMachinePool("pool1"),
+						InfraMachinePool: getLinuxAzureMachinePool("pool1"),
```

Comment:
this helper func `getLinuxAzureMachinePool` doesn't actually exist

Reply:
... ya i dont know how that happened. i found them in my local history and will commit